pyproject.toml

```diff
@@ -20,18 +20,25 @@ requires-python = ">=3.11,<3.12"
 
 dependencies = [
     # Core
-    "click==8.1.7",
-    "torch==2.1.2",
-    "scipy==1.10.1",
-    "wandb==0.16.3",
-    "protobuf==3.20.2",
+    "click>=8.1.7",
+    "scipy>=1.10.1",
+    "wandb>=0.16.3",
+    "protobuf>=3.20.2",
     "urllib3>=1.26.18,<2",
+    "pydantic>=2.6.4",
+    "pydantic-yaml>=1.2.0",
     "pydantic==2.6.4",
     "pydantic-yaml==1.2.0",
     "ray[default]==2.30.0",
     "loguru==0.7.2",
     "s3fs",
-    # HuggingFace
+]
+
+[project.optional-dependencies]
+ruff = ["ruff==0.2.1"]
+torchy = [
```

Comment:
maybe `jobs`  since it's `evaluation` + `finetuning`?

Reply:
i suppose that is indeed better than `torchy`. :)